documentation/behind-the-scenes-of-adding-a-file.md

```diff
@@ -0,0 +1,233 @@
+    UploadCAR[FP: Upload CAR to SP<br/><br/>✓ SP /piece/$pieceCid retrieval]
+    IndexCAR{{SP: Index CAR CIDs<br/><br/>✓ SP /ipfs/$cid retrieval}}
+    AdvertiseCAR{{SP: Advertise CAR CIDs to IPNI}}
+    AwaitIPNIIndexing[FP: Await IPNI Indexing<br/><br/>✓ IPNI provider records<br/>✓ IPFS Mainnet retrieval possible]
+    RetrieveData(Any: Retrieve with IPFS Mainnet<br/><br/>✓ ipfs://$cid works)
+
+    %% Blockchain nodes
+    ConnectWallet([User: Connect Wallet<br/><br/>✓ Wallet balances visible])
+    SetupPay[FP: Setup Filecoin Pay Account<br/><br/>✓ Filecoin Pay balance visible]
+    IdentifyDataSet[FP: Identify Data Set SP and ID]
+    CreateDataSet{{SP: Create Data Set & Add Piece<br/><br/>✓ Blockchain Transaction}}
+    ConfirmBlockchainTx[FP: Confirm Blockchain Transaction<br/><br/>✓ IPNI provider records<br/>✓ Data Set & Piece metadata onchain]
+    ProveData{{SP: Prove Data Possession<br/><br/>✓ Cryptographic proofs onchain and visible on explorers}}
+
+    %% Milestone
+    FilecoinPinAddDone{FP: `Add` Done}
+
+    %% Relationships
+    %% Non-blockchain flow
+    Start --> CreateCAR
+    CreateCAR --> UploadCAR
+    UploadCAR --> IndexCAR
+    IndexCAR --> AdvertiseCAR
+    AdvertiseCAR --> AwaitIPNIIndexing
+    AwaitIPNIIndexing --> FilecoinPinAddDone
+    FilecoinPinAddDone --> RetrieveData
+
+    %% Blockchain flow
+    ConnectWallet --> SetupPay
+    SetupPay --> IdentifyDataSet
+    IdentifyDataSet --> CreateDataSet
+
+    %% Convergence
+    UploadCAR --> CreateDataSet
+    CreateDataSet --> ConfirmBlockchainTx
+    ConfirmBlockchainTx --> FilecoinPinAddDone
+    FilecoinPinAddDone --> ProveData
+
+    %% Styling
+    classDef nonBlockchain fill:#e1f5ff,stroke:#0288d1,stroke-width:2px,color:#000
+    classDef blockchain fill:#fff3e0,stroke:#f57c00,stroke-width:2px,color:#000
+
+    class Start,CreateCAR,UploadCAR,IndexCAR,AdvertiseCAR,AwaitIPNIIndexing,RetrieveData nonBlockchain
+    class ConnectWallet,SetupPay,IdentifyDataSet,CreateDataSet,ConfirmBlockchainTx,ProveData blockchain
+```
+
+## Steps without Blockchain Interactions
+
+These are the set of steps that are done client side (i.e., where the Filecoin Pin code is running) and with a [Storage Provider](glossary.md#storage-provider) that don't involve the Filecoin blockchain.  These steps in isolation though don't yield a committed cryptographic proof of the data being possessed by and retrievable from an SP, but they are necessary preconditions.
+
+### Create CAR
+
+*What/why:*
+
+The provided file needs to be turned into a merkle DAG and have the DAG's blocks shipped off.  [CAR](glossary.md#car) is a common container format for transporting blocks in the IPFS ecosystem and is used with Filecoin Pin.  Filecoin Pin DAGifies the content so that it knows the CIDs of the blocks so it doesn't need to trust the [Storage Provider](glossary.md#storage-provider) (SP).
+
+Implementation notes:
+- The CAR file is created using Helia for UnixFS DAG creation.
```

Comment:
idk if we care about diving into the details, but we do have some car creation nuance

Reply:
I'll wait to add more details until it gets asked about more.